src/Request.php

```diff
@@ -141,22 +145,24 @@ public function handleData($data)
             $this->stream->removeListener('end', array($this, 'handleEnd'));
             $this->stream->removeListener('error', array($this, 'handleError'));
 
-            $this->response = $response;
+            if (isset($response)) {
```

Comment:
Functionally LGTM, but the changeset could be significantly smaller by using an early return above :+1: